src/osd/OSD.cc

```diff
@@ -7569,6 +7569,18 @@ void OSD::sched_scrub()
 {
   auto& scrub_scheduler = service.get_scrub_services();
 
+  if (auto blocked_pgs = scrub_scheduler.get_blocked_pgs_count();
+      blocked_pgs > 0) {
+    // some PGs managed by this OSD were blocked by a locked object during
+    // scrub. This means we might not have the resources needed to scrub now.
+    dout(10)
+      << fmt::format(
+	   "{}: PGs are blocked while scrubbing due to locked objects ({} PGs)",
+	   __func__,
+	   blocked_pgs)
```

Comment:
Perhaps it's worth indicating which PGs are blocked. meaning, not only how many of them.

Reply:
I've considered that. but wasn't sure the added complexity (well - a std::set and
a few lines more in the OSD code) is worth it. After all - once noticing the problem,
one is expected to examine the cluster log etc.

Reply:
The OSD code can remain untouched, with `ScrubQueue::mark_pg_scrub_blocked(spg_t blocked_pg)` we can use `blocked_pg` for the set.

Reply:
https://github.com/ceph/ceph/pull/46643#discussion_r899119719